File: acmp.ru/0619/619.cpp
#include<iostream>
#include<iomanip>
#include<vector>

using namespace std;

int main()
{
    int n, sum;

    cin >> n >> sum;

    vector<double> dpCur(n*6+1,0), dpNext(n*6+1,0);

    dpCur[0] = 1;

    for(int i = 0; i<n; i++)
    {
        for(int j = i; j<=i*6; j++)
            for(int delta = 1; delta<=6; delta++)
                dpNext[j+delta]+=dpCur[j]/6;
        dpNext.swap(dpCur);
        dpNext.assign(n*6+1, 0);
    }
    cout.precision(10);
    cout << dpCur[sum];
    return 0;
}

File: acmp.ru/0619/619.py
N, total = map(int, input().split(' '))


dpCur = [0]*(N*6+1)

dpCur[0] = 1

for i in range(N):
    dpNext = [0]*(N*6+1)
    for j in range(i*6+1):
        for delta in range(1,7):
            dpNext[j+delta]+=dpCur[j]/6
    dpCur = dpNext
print(dpCur[total])
